### PR TITLE
[MM-13381] Added reply to mobile longpress post options

### PR DESCRIPTION
**Pull request by courtneypattison**
#### Summary
Adds reply to post options. This is my first time using React Native, so apologies in advance for my ignorance!

#### Ticket Link
Jira ticket: [MM-13381](https://mattermost.atlassian.net/browse/MM-13381)
Fixes: mattermost/mattermost-server#10191

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file updates

#### Device Information
This PR was tested on: iPhone XR Simulator, 10.3  and Pixel 2 Simulator, 28 

#### Screenshots
screen shot 2019-02-28 at 10 48 53 am
screen shot 2019-02-28 at 10 49 13 am
screen shot 2019-02-28 at 10 48 31 am
screen shot 2019-02-28 at 10 31 54 am
screen shot 2019-02-28 at 10 32 32 am
screen shot 2019-02-28 at 10 33 20 am

### CODE REVIEW

File: app/screens/post_options/index.js [hunk 1]
```diff
@@ -39,6 +41,7 @@ function mapStateToProps(state, ownProps) {
     const channelIsArchived = channel.delete_at !== 0;
 
     let canAddReaction = true;
+    let canReply = true;
```
Comment (enahum) [inline app/screens/post_options/index.js]: What if you are already in the thread view? this option should not be included if that is the case

File: app/screens/post_options/post_option.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ import flag from 'assets/images/post_menu/flag.png';
 import link from 'assets/images/post_menu/link.png';
 import pin from 'assets/images/post_menu/pin.png';
 import trash from 'assets/images/post_menu/trash.png';
+import reply from 'assets/images/post_menu/reply.png';
```
Comment (enahum) [inline app/screens/post_options/post_option.js]: perhaps we should remove this component `app/components/reply_icon` as we now have assets for it, and lets take into account that the current reply_icon component is an svg which is not as performant as an asset.

thoughts?

File: app/screens/post_options/post_options.js [hunk 3]
```diff
@@ -205,56 +225,21 @@ export default class PostOptions extends PureComponent {
...
-        }
-        if (canDelete) {
-            actions.push(this.getDeleteOption());
-        }
-
-        return actions.filter((a) => a !== null);
-    };
-
-    getOthersPostOptions = () => {
+    getPostOptions = () => {
```
Comment (enahum) [inline app/screens/post_options/post_options.js]: there is a reason why we have `getMyPostOptions` and `getOthersPostOptions` the options display order is different for the two, please revert this change.

File: app/screens/post_options/post_options.js [hunk 4]
```diff
@@ -287,6 +272,41 @@ export default class PostOptions extends PureComponent {
         actions.addReaction(post.id, emoji);
     };
 
+    handleReply = () => {
+        this.goToThread();
```
Comment (enahum) [inline app/screens/post_options/post_options.js]: `goToThread` is not needed, you can just use this function and include the code in the `goToThread` one

File: app/screens/post_options/post_options.js [hunk 5]
```diff
@@ -287,6 +272,41 @@ export default class PostOptions extends PureComponent {
...
+    }
+
+    goToThread = () => {
+        const {actions, post, navigator, theme} = this.props;
+        const rootId = (post.root_id || post.id);
+        const channelId = post.channel_id;
+
+        actions.loadThreadIfNecessary(post.root_id, channelId);
```
Comment (enahum) [inline app/screens/post_options/post_options.js]: shouldn't this be

```suggestion
        actions.loadThreadIfNecessary(rootId, channelId);
```

File: assets/base/i18n/de.json [hunk 6]
```diff
@@ -334,6 +334,7 @@
   "mobile.pinned_posts.empty_description": "Wichtige Elemente anheften durch gedrückt halten einer Nachricht und wählen von \"An Kanal anheften\".",
   "mobile.pinned_posts.empty_title": "Keine angehefteten Nachrichten",
   "mobile.post_info.add_reaction": "Reaktion hinzufügen",
+  "mobile.post_info.reply": "Antworten",
```
Comment (enahum) [inline assets/base/i18n/de.json]: We only need the addition in `en.json` the rest of the languages will be taken care by https://docs.mattermost.com/developer/localization.html#translations-updates

Comment (enahum) [reply assets/base/i18n/de.json]: also we can just use `post_info.reply` as it already exists and there is no need for yet another key ;)

File: app/screens/post_options/index.js (hunk 1 above)
Comment (courtneypattison) [reply app/screens/post_options/index.js]: Good catch!

File: app/screens/post_options/post_options.js (hunk 3 above)
Comment (courtneypattison) [reply app/screens/post_options/post_options.js]: Sorry! I was trying to refactor and messed up. What are the orders supposed to be? Does it make sense to get rid of the splice and push and just have the order in the actions array? Something like this but with the proper order:
```
getOthersPostOptions = () => {
        const actions = [
            this.getAddReactionOption(),
            this.getReplyOption(),
            this.getCopyText(),
            this.getEditOption(),
            this.getFlagOption(),
            this.getPinOption(),
            this.getCopyPermalink(),
            this.getDeleteOption(),
        ];

        return actions.filter((a) => a !== null);
    };

    getMyPostOptions = () => {
        const actions = [
            this.getAddReactionOption(),
            this.getReplyOption(),
            this.getCopyText(),
            this.getEditOption(),
            this.getFlagOption(),
            this.getPinOption(),
            this.getCopyPermalink(),
            this.getDeleteOption(),
        ];

        return actions.filter((a) => a !== null);
    };

    getPostOptions = () => {
        const {isMyPost} = this.props;

        return isMyPost ? this.getMyPostOptions() : this.getOthersPostOptions();
    };
```

File: app/screens/post_options/post_options.js (hunk 4 above)
Comment (courtneypattison) [reply app/screens/post_options/post_options.js]: Okie doke :)

File: app/screens/post_options/post_options.js (hunk 5 above)
Comment (courtneypattison) [reply app/screens/post_options/post_options.js]: Whoops

File: assets/base/i18n/de.json (hunk 6 above)
Comment (courtneypattison) [reply assets/base/i18n/de.json]: Just to make sure I'm understanding correctly, should I remove all the translations and just keep the `"mobile.post_info.reply": "Reply",` en.json?

File: app/screens/post_options/post_options.js (hunk 3 above)
Comment (enahum) [reply app/screens/post_options/post_options.js]: that would make sense yes.

The order is in the [spec](https://docs.google.com/document/d/1CZwdcPXTP9I3XqTbz67jomJ912TQrWF8BUr1BioEPuU/). If you don't have access to the doc feel free to request access to it.

File: assets/base/i18n/de.json (hunk 6 above)
Comment (enahum) [reply assets/base/i18n/de.json]: That is correct

File: app/screens/post_options/post_options.js (hunk 3 above)
Comment (courtneypattison) [reply app/screens/post_options/post_options.js]: Thank you for the information! I've requested access and will update the PR once I get it.

**Comment by esethna** [comment]
@courtneypattison let us know if you need any help clarifying on the above feedback?

**Comment by courtneypattison** [comment]
@esethna The feedback was great! The updates are ready :) I'm just waiting for permission to access the spec so I can verify the post options order. I'll commit the changes once I know the order.

**Comment by enahum** [comment]
@esethna did you grant access to @courtneypattison on the post menu spec?

**Comment by enahum** [comment]
@courtneypattison do you think this will be ready by Tuesday 26th of March? if not we are going to be force to push it to the 1.20 release

**Comment by courtneypattison** [comment]
@enahum I'm still just waiting on permission to view the spec. Is it possible for you to send me the order for the post options? Or I could just push my updates with a guess at the order so at least the other updates can be reviewed, then I could fix the order later.

**Comment by enahum** [comment]
@courtneypattison the order should remain the same as it currently is and the Reply option should be:

* Own posts: Reply is the second option
* Others posts: Reply is the first option

**Comment by courtneypattison** [comment]
I'm so sorry :( I messed up my commit with some accidental merging. I'll try it fix it tonight when my son goes to sleep!

**Comment by courtneypattison** [comment]
Couldn't figure out how to fix the history, sorry! New [PR](https://github.com/mattermost/mattermost-mobile/pull/2610).

**Comment by enahum** [comment]
No worries, lets close this PR in favor of the new one
